### PR TITLE
Relayer logging

### PR DESCRIPTION
**Pull request by drinkcoffee**
* Added relayer component.
* Added relayer build and unit test to Circle CI.

### CODE REVIEW

File: messaging/relayer/internal/logging/logging_test.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	Info("info")
+	Warn("warn")
+	Error("error")
+	// Output:
+}
+
+func Example_setLogLevel() {
+	conf := viper.New()
+	conf.Set("LOG_LEVEL", "AA")
+	setLogLevel(conf)
```
Comment (ermyas) [inline messaging/relayer/internal/logging/logging_test.go:63]: is there a specific purpose for lines 62 and 63 as part of this test? Is the intent to show that the subsequent `setLogLevel()` would override this setting? If so, might it make sense to instead use a valid log level that would not be ignored in place of "AA" e.g. `info` .. alternatively, if the test is to ensure that an unknown level would be ignored, should we add a check for that?

**Review by ermyas — APPROVED** [review]
LGTM added a couple of minor questions/comments.

File: messaging/relayer/internal/logging/logging.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	if err != nil {
+		log.Error().Err(err).Msg("can't parse log level")
+		level = zerolog.InfoLevel
+		log.Warn().Msgf("using level '%v' as default", level)
+	}
+	zerolog.SetGlobalLevel(level)
+}
+
+// SetLogLevel sets the logging level.
+func SetLogLevel(logLevel string) {
```
[the remark this answers is not in the document]
Comment (drinkcoffee) [reply messaging/relayer/internal/logging/logging.go]: Now returns an error.

File: messaging/relayer/internal/logging/logging_test.go (hunk 1 above)
Comment (drinkcoffee) [reply messaging/relayer/internal/logging/logging_test.go:67]: Fixed the test.

**Pull request merged**